### PR TITLE
Merge metrics from 2.x

### PR DESCRIPTION
**Pull request by bwarden**
Merging existing network metrics tests from the 2.x repository, preserving history. Moving from "tests/metrics/network" to "metrics/network", since this repository is already named "tests". No other porting changes included.

Signed-off-by: Brett T. Warden <brett.t.warden@intel.com>

### CODE REVIEW

**Comment by chavafg** [comment]
Not sure if this is going to be a 2-step process, but some of the files here (those with `.in`) were processed by `autotools` and we do not have that here, so they will not work. 

If that is going to be fixed in another PR, can you please create an issue to track it?
Thanks.

**Comment by bwarden** [comment]
This will be multiple steps. First, I'm bringing the existing files over, then I'll make each one work.
- Rework generated files: #170

**Comment by bwarden** [comment]
This is the first step toward resolving #149

**Comment by grahamwhaley** [comment]
tbh, I would have preferred that we fixed the files as we imported them, otherwise we end up placing a bunch of 'known broken' files in our tree, and generally that is not our workflow.
I do like you keeping the history of the patches there though @bwarden 
Would you be amenable to adding your patch(es) on top of the history before we do the PR merge @bwarden , so we don't end up with a bunch of broken code in the tree (albeit temporarily).

**Comment by bwarden** [comment]
Alright, so you'd like me to push patches that implement patching the tests in this same PR? Should work. I'll keep each patch to a logical set of changes, of course.

**Comment by grahamwhaley** [comment]
Thanks. I think that is going to allow us to land a better first set of code (rather than something known to not work).
It it works better for you then you could do the tests individually or in smaller batches. See how 'big and complex' it feels. Smaller bits can bring some benefits:
 - easier (and thus more likely to get) review
 - less inter-dependence - if one part blocks it does not block the other PRs.

File: metrics/network/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# Networking tests for the Intel® Clear Containers runtime
```
Comment (iphutch) [inline metrics/network/README.md]: "the" is not necessary here. 
Should read: Networking tests for Intel® Clear Containers runtime

File: metrics/network/README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+Currently, the Clear Containers have a series of network performance tests.
```
Comment (iphutch) [inline metrics/network/README.md]: remove/the

File: metrics/network/README.md (hunk 1 above)
Comment (bwarden) [inline metrics/network/README.md]: Will remove

File: metrics/network/README.md (hunk 2 above)
Comment (bwarden) [inline metrics/network/README.md]: Will remove

**Comment by bwarden** [comment]
Travis doesn't like this commit (Commit 5ca8d209d517fc75531cf5a34d82357bef75c933: Failed to find subsystem in subject: "Dockerfile for network image"):
`
commit 5ca8d209d517fc75531cf5a34d82357bef75c933
Author: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>
Date:   Tue May 9 19:13:18 2017 +0000

    Dockerfile for network image
    
    This Dockerfile contains all the tools and dependencies
    requiered to run the network tests.
    
    Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>
`

**Comment by bwarden** [comment]
Moved qemu-finding function change to separate PR #182

**Comment by grahamwhaley** [comment]
Hi @bwarden Thanks for fixing the things from the feedback.
I'm guessing you have more patches to come, such as adding the `save_results` call to all the tests etc.
If so, we should mark this PR as work in progress (WIP) or do not merge, probably best by adding the relevant github label to this PR. That way we don't run the risk of somebody accidentally merging or giving a full 'ack' until the series is complete.
On that note - you can push patches up to this PR whilst still working on the port, but note, each time you push it notifies us and we will come do a review (at least read the new code), so you may want to batch any changesets up to nice logical bits to review - both for uniformity and to not burn out the reviewers :-)

**Comment by bwarden** [comment]
Once the next force push succeeds, review solicited. Commit 52fe8167497d21d3a3bdfb0882ac5bfcb6560a98 focuses on putting the necessary common functions in place for the network metrics to actually run. I have not verified that the tests are doing the right thing yet.

File: metrics/lib/common.bash [hunk 3]
```diff
@@ -18,12 +18,17 @@ SCRIPT_PATH=$(dirname $(readlink -f $0))
 RESULT_DIR="${SCRIPT_PATH}/../results"
 LIB_DIR="${SCRIPT_PATH}/../lib"
 
+# Set variables to reasonable defaults if unset or empty
+DOCKER_EXE="${DOCKER_EXE:-docker}"
+RUNTIME="${RUNTIME:-cor}"
```
Comment (grahamwhaley) [inline metrics/lib/common.bash]: maybe set the default to the CC3.x runtime name rather than the CC2.x runtime name.

Comment (bwarden) [inline metrics/lib/common.bash]: Changed to "cc-runtime" and pushed.

File: metrics/lib/common.bash [hunk 4]
```diff
@@ -84,74 +89,105 @@ function init_env()
...
+	else
+		return 0
+	fi
+}
+
+# This function checks if there are containers or
+# cc-shim and qemu-lite processes up, if found, they are
+# killed to start test with clean environment.
+# The function is intended to be used in the setup() of
+# the docker integration tests for cc-oci-runtime.
```
Comment (jodh-intel) [inline metrics/lib/common.bash]: Mention of `cc-oci-runtime`. Does this comment need updating?

File: metrics/network/dockerfile_image/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,40 @@
+#  This file is part of cc-oci-runtime.
```
Comment (jodh-intel) [inline metrics/network/dockerfile_image/Dockerfile]: Comment is now incorrect.

File: metrics/network/lib/network-test-common.bash [hunk 6]
```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+#  This file is part of cc-oci-runtime.
```
Comment (jodh-intel) [inline metrics/network/lib/network-test-common.bash]: Comment is now incorrect.

File: metrics/network/wip/network-latency.sh [hunk 7]
```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+#  This file is part of cc-oci-runtime.
```
Comment (jodh-intel) [inline metrics/network/wip/network-latency.sh]: Comment is now incorrect.

File: metrics/network/wip/network-metrics-cpu-consumption.sh [hunk 8]
```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+#  This file is part of cc-oci-runtime.
```
Comment (jodh-intel) [inline metrics/network/wip/network-metrics-cpu-consumption.sh]: Comment is now incorrect.

File: metrics/network/wip/network-metrics-iperf3.sh [hunk 9]
```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+#  This file is part of cc-oci-runtime.
```
Comment (jodh-intel) [inline metrics/network/wip/network-metrics-iperf3.sh]: Comment is now incorrect.

File: metrics/network/wip/network-metrics-memory-pss-1g.sh [hunk 10]
```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+#  This file is part of cc-oci-runtime.
```
Comment (jodh-intel) [inline metrics/network/wip/network-metrics-memory-pss-1g.sh]: Comment is now incorrect.

File: metrics/network/wip/network-metrics-memory-pss.sh [hunk 11]
```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+#  This file is part of cc-oci-runtime.
```
Comment (jodh-intel) [inline metrics/network/wip/network-metrics-memory-pss.sh]: Comment is now incorrect.

File: metrics/network/wip/network-metrics-memory-rss-1g.sh [hunk 12]
```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+#  This file is part of cc-oci-runtime.
```
Comment (jodh-intel) [inline metrics/network/wip/network-metrics-memory-rss-1g.sh]: Comment is now incorrect.

File: metrics/network/wip/network-metrics-nuttcp.sh [hunk 13]
```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+#  This file is part of cc-oci-runtime.
```
Comment (jodh-intel) [inline metrics/network/wip/network-metrics-nuttcp.sh]: Comment is now incorrect.

File: metrics/network/wip/network-metrics.sh [hunk 14]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+#  This file is part of cc-oci-runtime.
```
Comment (jodh-intel) [inline metrics/network/wip/network-metrics.sh]: Comment is now incorrect.

File: metrics/network/wip/network-nginx-ab-benchmark.sh [hunk 15]
```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+#  This file is part of cc-oci-runtime.
```
Comment (jodh-intel) [inline metrics/network/wip/network-nginx-ab-benchmark.sh]: Comment is now incorrect.

**Comment by grahamwhaley** [comment]
As an update from an offline discussion - to ease the process of updating and landing all the tests we are going to initially land much of this code in a form that won't function properly (hence, items now placed under the `wip` folder), and then do individual updates to enable and move tests into their working place and state one at a time. Right now that is going to be the most efficient workflow to get all the tests landed and working.
Thus, @bwarden please review @jodh-intel comments for any parts not in the wip folder. For the parts in wip, thanks for the feedback @jodh-intel We may initially land with some of that still in place, and pick up the feedback as we do the subsequent PRs to update.
So, sure, in a perfect world I'd rather not land things in tree that don't actually function, but here we are trying to be prudently efficient.
I guess an alternative flow would (have been) a topic branch. I'm happy to land this PR (post a few more edits) though and push on.

File: metrics/network/dockerfile_image/Dockerfile (hunk 5 above)
Comment (grahamwhaley) [inline metrics/network/dockerfile_image/Dockerfile]: @jodh-intel you could have just said 'and in a number of other files, please grep and fix' :-)

**Comment by bwarden** [comment]
I've removed all comments referring to cc-oci-runtime. I've also adjusted the default CC_SHIM_PATH value to /usr/libexec/clear-containers/cc-shim. This is only used in a function that looks for running processes to kill to ensure the test starts with a clean slate (kill_processes_before_start in metrics/lib/common.bash). Do I need to add cc-proxy to that function as well?

**Comment by grahamwhaley** [comment]
Hi @bwarden for the process reset/clean/kill - the proxy is/would have to be treated a little differently as it is something owned/managed by systemd. There is already some code in that file that uses `systemctl` do do something similar, and we should probably examine that code and amend to make sure it also does the correct thing for the `cc-runtime` runtime:
https://github.com/clearcontainers/tests/blob/master/metrics/lib/common.bash#L81

**Comment by grahamwhaley** [comment]
Thanks @bwarden I'm going to merge this. If I were being picky then I'd probably ask for some patch shuffle and merging, like we could do all the file name/moves in a single patch etc. - but, let's get this in and then work on fixing the tests up into the final positions.

**Pull request merged**